File: foodguide/src/main/java/com/vinhkhanh/foodguide/entity/Location.java
package com.vinhkhanh.foodguide.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Table(name = "location")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Location {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "location_id")
    private Long locationId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "latitude", nullable = false)
    private Double latitude;

    @Column(name = "longitude", nullable = false)
    private Double longitude;

    @Column(name = "image_url")
    private String imageUrl;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "location", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Food> foods;

    @OneToMany(mappedBy = "location", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<AudioGuide> audioGuides;

    @OneToMany(mappedBy = "location", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<QRCode> qrCodes;
}

File: vinh-khanh-food-guide/shared/models/Location.ts
/**
 * Shared Location Model - Compatible with Spring Boot Backend
 */

export interface Location {
  locationId: number;
  name: string;
  description?: string;
  latitude: number;
  longitude: number;
  imageUrl?: string;
  userId?: number;
  foods?: Food[];
  audioGuides?: AudioGuide[];
  qrCode?: QRCode;
}

export interface LocationDetail extends Location {
  foods: Food[];
  audioGuides: AudioGuide[];
  qrCode: QRCode;
}

export interface LocationRequest {
  name: string;
  description?: string;
  latitude: number;
  longitude: number;
  imageUrl?: string;
}
